src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncReceiptsRequestHandler.scala

```diff
@@ -32,6 +37,22 @@ class FastSyncReceiptsRequestHandler(
     cleanupAndStop()
   }
 
+  private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
+    val fullBlocks = receivedHashes.flatMap { hash =>
+      for {
+        header <- blockchain.getBlockHeaderByHash(hash)
+        _ <- blockchain.getBlockBodyByHash(hash)
+      } yield header
+    }
+
+    if (fullBlocks.nonEmpty) {
+      val bestReceivedBlock = fullBlocks.maxBy(_.number)
```

Comment:
it is possible. however, see my comment below. the implementation is going to change